[bug] CMake can't resolve libraries of conan editable packages with subprojects

### Environment details

* Operating System+version: `Linux host 6.1.19-gentoo-x86_64`
* Compiler+version: `gcc (Gentoo 12.2.1_p20230428-r1 p2) 12.2.1`
* Conan version: `Conan version 2.0.6`
* Python version: `Python 3.11.3`


### Steps to reproduce

####  Description

I tried to rewrite example [`examples2/tutorial/developing_packages/editable_packages`](https://github.com/conan-io/examples2/tree/24aff62812b44262f88f122b695a62c333b46142/tutorial/developing_packages/editable_packages) using submodules hello and say, but when I did it CMake stopped resolving libraries frommodules on which it depends (from say module in case of cmake cache generation for hello).

#### Configuration

`say/conanfile.py`:

```python
from conan import ConanFile
from conan.tools.cmake import CMakeToolchain, CMake, cmake_layout

class SayConan(ConanFile):
    name = "say"
    version = "1.0"

    # Binary configuration
    settings = "os", "compiler", "build_type", "arch"
    options = {"shared": [True, False], "fPIC": [True, False]}
    default_options = {"shared": False, "fPIC": True}

    # Sources are located in the same place as this recipe, copy them to the recipe
    exports_sources = "CMakeLists.txt", "src/*", "include/*"

    _folders_rel_root = ".."
    _folders_rel_sub = "say"

    def config_options(self):
        if self.settings.os == "Windows":
            del self.options.fPIC

    def layout(self):
        self.folders.root = self._folders_rel_root
        self.folders.subproject = self._folders_rel_sub
        cmake_layout(self)

    def generate(self):
        tc = CMakeToolchain(self)
        tc.generate()

    def build(self):
        cmake = CMake(self)
        cmake.configure()
        cmake.build()

    def package(self):
        cmake = CMake(self)
        cmake.install()

    def package_info(self):
        self.cpp_info.libs = ["say"]
```

`hello/conanfile.py`:

```python
from conan import ConanFile
from conan.tools.cmake import cmake_layout

class HelloConan(ConanFile):
    name = "hello"
    version = "1.0"

    settings = "os", "compiler", "build_type", "arch"

    generators = "CMakeToolchain", "CMakeDeps"
    requires = "say/1.0"

    _folders_rel_root = ".."
    _folders_rel_sub = "hello"

    def layout(self):
        self.folders.root = self._folders_rel_root
        self.folders.subproject = self._folders_rel_sub
        cmake_layout(self)
```

#### Steps

1. Go to root of current example
2. Add say to editable `conan editable add say`
3. Execute in order for say:

```sh
cd say
conan install .
cmake --preset conan-release
cmake --build --preset conan-release
cd ..
```

4. Now try the same for hello:

```sh
cd hello
conan install .
cmake --preset conan-release # <- cmake error
```

### Logs

```
(conan) nikit@host hello % cmake --preset conan-release
Preset CMake variables:

  CMAKE_BUILD_TYPE="Release"
  CMAKE_POLICY_DEFAULT_CMP0091="NEW"
  CMAKE_TOOLCHAIN_FILE:FILEPATH="/home/nikit/Workspaces/research/conan/simple/examples2/tutorial/developing_packages/editable_packages/hello/build/Release/generators/conan_toolchain.cmake"

-- Using Conan toolchain: /home/nikit/Workspaces/research/conan/simple/examples2/tutorial/developing_packages/editable_packages/hello/build/Release/generators/conan_toolchain.cmake
-- Conan toolchain: C++ Standard 20 with extensions OFF
-- Conan: Target declared 'say::say'
CMake Error at build/Release/generators/cmakedeps_macros.cmake:64 (message):
  Library 'say' not found in package.  If 'say' is a system library, declare
  it with 'cpp_info.system_libs' property
Call Stack (most recent call first):
  build/Release/generators/say-Target-release.cmake:24 (conan_package_library_targets)
  build/Release/generators/sayTargets.cmake:26 (include)
  build/Release/generators/say-config.cmake:16 (include)
  CMakeLists.txt:5 (find_package)
  ```

--- a/conans/client/installer.py
+++ b/conans/client/installer.py
@@ -343,8 +343,9 @@ class BinaryInstaller:
         conanfile_path = editable["path"]
         output_folder = editable.get("output_folder")
 
-        # TODO: Check, this assumes the folder is always the conanfile one
         base_path = os.path.dirname(conanfile_path)
+        base_path = base_path if conanfile.folders.root is None else \
+            os.path.normpath(os.path.join(base_path, conanfile.folders.root))
         conanfile.folders.set_base_folders(base_path, output_folder)
         output = conanfile.output
         output.info("Rewriting files of editable package "

--- a/conans/test/integration/editable/test_editable_layout.py
+++ b/conans/test/integration/editable/test_editable_layout.py
@@ -1,0 +1,36 @@
+import os
+import textwrap
+
+from conans.test.utils.tools import TestClient
+
+
+def test_editable_folders_root():
+    """ Editables with self.folders.root = ".." should work too
+    """
+    c = TestClient()
+    sub1 = textwrap.dedent("""
+        from conan import ConanFile
+
+        class Pkg(ConanFile):
+            name = "pkg"
+            version = "0.1"
+            def layout(self):
+                self.folders.root = ".."
+                self.folders.build = "mybuild"
+                self.cpp.build.libdirs = ["mylibdir"]
+        """)
+    sub2 = textwrap.dedent("""
+        from conan import ConanFile
+
+        class Pkg(ConanFile):
+            requires = "pkg/0.1"
+            def generate(self):
+                pkg_libdir = self.dependencies["pkg"].cpp_info.libdir
+                self.output.info(f"PKG LIBDIR {pkg_libdir}")
+        """)
+    c.save({"sub1/conanfile.py": sub1,
+            "sub2/conanfile.py": sub2})
+    c.run("editable add sub1")
+    c.run("install sub2")
+    libdir = os.path.join(c.current_folder, "mybuild", "mylibdir")
+    assert f"conanfile.py: PKG LIBDIR {libdir}" in c.out

